fragments/master.sh

```diff
@@ -209,74 +188,52 @@ start_k8s(){
             --allow-privileged=true --v=2 \
-            --pod-infra-container-image=${PAUSE_IMAGE}
-}
-
-run_addons_container(){
-  docker run \
-      --net=host \
-      -d \
-      ${ADDONS_IMAGE}
-}
-
-set +e
-deploy_add_ons(){
-  # poll the server until it starts up then run addons
-  wait_on_api_server
-  run_addons_container
+            --pod-infra-container-image=${PAUSE_IMAGE} \
+            --network-plugin=cni \
+            --network-plugin-dir=/etc/cni/net.d
 }
 
-wait_on_api_server(){
-  # init vars
-  which curl > /dev/null || {
-    echo "curl must be installed"
-    exit 1
-  }
-
-  local URL='http://localhost:8080/healthz'
-  local RESPONSE=$(curl --write-out %{http_code} --silent --output /dev/null $URL )
-  local OK_STATUS=200
-  local TIMEOUT=300 # 5 minutes
-  local INTERVAL=5 # every 5 seconds
-
-  local NEXT_WAIT_TIME=0
-  until [ ${RESPONSE:-0} -eq $OK_STATUS ] || [ $NEXT_WAIT_TIME -eq $TIMEOUT ]; do
-    sleep ${INTERVAL}
-    RESPONSE=$(curl --write-out %{http_code} --silent --output /dev/null $URL )
-    let NEXT_WAIT_TIME=NEXT_WAIT_TIME+INTERVAL
-  done
-
-  if [ ${RESPONSE:-0} -eq $OK_STATUS ]; then
-        echo "Success: Response 200 received from API service ..."
-  elif [ $NEXT_WAIT_TIME -eq $TIMEOUT ]; then
-        echo "Error: Timeout has occurred while trying to contact the API service ..."
-  else
-        echo "Error: API service cannot be reched ..."
-  fi
-}
-
-set_docker_registry(){
-
-	# Add docker registry as prefix for k8s images.
-	if [[ -n ${DOCKER_REGISTRY_PREFIX} ]]; then
-	  ETCD_IMAGE=${DOCKER_REGISTRY_PREFIX}/${ETCD_IMAGE}
-	  FLANNEL_IMAGE=${DOCKER_REGISTRY_PREFIX}/${FLANNEL_IMAGE}
-	  HYPERKUBE_IMAGE=${DOCKER_REGISTRY_PREFIX}/${HYPERKUBE_IMAGE}
-	  ADDONS_IMAGE=${DOCKER_REGISTRY_PREFIX}/${ADDONS_IMAGE}
-	  PAUSE_IMAGE=${DOCKER_REGISTRY_PREFIX}/${PAUSE_IMAGE}
-	fi
-
+configure_docker(){
+	# Configure docker settings, then restart it
     case "${lsb_dist}" in
+        amzn)
+            DOCKER_CONF="/etc/sysconfig/docker"
+            sed -i.bak 's/^\(MountFlags=\).*/\1shared/' $DOCKER_CONF
+            service docker restart
+            ;;
         centos)
-			DOCKER_CONF="/usr/lib/systemd/system/docker.service"
-			DOCKER_BOOTSTRAP_CONF="/usr/lib/systemd/system/docker-bootstrap.service"
-			if [[ -n ${DOCKER_REGISTRY_URL} ]]; then
+            DOCKER_CONF="/usr/lib/systemd/system/docker.service"
+            sed -i.bak 's/^\(MountFlags=\).*/\1shared/' ${DOCKER_CONF}
+            # Add support for docker registry
+            if [[ -n ${DOCKER_REGISTRY_URL} ]]; then
 			  sed -i "/^ExecStart=/ s~$~ --insecure-registry=${DOCKER_REGISTRY_URL}~" ${DOCKER_CONF}
-			  sed -i "/graph=/ s~$~ --insecure-registry=${DOCKER_REGISTRY_URL}~" ${DOCKER_BOOTSTRAP_CONF}
-			  systemctl daemon-reload
-			  systemctl restart docker
-			  systemctl restart docker-bootstrap
 			fi
+			# Add docker registry as prefix for k8s images.
+			if [[ -n ${DOCKER_REGISTRY_PREFIX} ]]; then
```

Comment:
I would expect such definition in a function at the top of the file. Similar the way luxas did it.

maybe a function called init() or something